File: src/codes/single-state-handling/ts/ksrc.ts
class SupportTicket {
  private status: string = "open"; // 状態: open, in_progress, resolved, closed

  handle() {
    if (this.status === "open") {
      console.log("チケットを処理中に設定します");
      this.status = "in_progress";
    } else if (this.status === "in_progress") {
      console.log("チケットを解決済みに設定します");
      this.status = "resolved";
    } else if (this.status === "resolved") {
      console.log("チケットをクローズします");
      this.status = "closed";
    } else if (this.status === "closed") {
      console.log("このチケットはすでに終了しています");
    }
  }
}

File: src/codes/single-state-handling/py/ksrc.py
class SupportTicket:
    def __init__(self):
        self.status = "open"  # 状態: open, in_progress, resolved, closed

    def handle(self):
        if self.status == "open":
            print("チケットを処理中に設定します")
            self.status = "in_progress"
        elif self.status == "in_progress":
            print("チケットを解決済みに設定します")
            self.status = "resolved"
        elif self.status == "resolved":
            print("チケットをクローズします")
            self.status = "closed"
        elif self.status == "closed":
            print("このチケットはすでに終了しています")
